src/components/ImageCarousel/ImageCarousel.tsx

```diff
@@ -44,29 +81,123 @@ const ImageCarousel = ({
-        {images.map((_, index) => (
+      {navType === "arrows" && (
+        <>
           <button
-            key={index}
-            className={`
+            className="absolute bottom-0 top-0 block text-6xl"
+            onClick={showPrevImage}
+          >
+            <VscTriangleLeft />
+          </button>
+          <button
+            className="absolute bottom-0 right-0 top-0 block text-6xl"
+            onClick={showNextImage}
+          >
+            <VscTriangleRight />
+          </button>
+        </>
+      )}
+
+      {navType === "circles" && (
+        <div className="absolute bottom-2 left-1/2 flex -translate-x-1/2 transform gap-1">
+          {images.map((_, index) => (
+            <button
+              key={index}
+              className={`
             relative mx-[6px] inline-block h-[9px] w-[9px] cursor-pointer rounded-full border [transition:color_0.2s_ease-in-out]
               ${
                 index === imageIndex
-                  ? "border-yellow-400 bg-yellow-400"
+                  ? "border-[#d9c6c5] bg-[#d9c6c5]"
                   : "border-[#d9c6c5] bg-transparent"
               }
             `}
-            onClick={() => setImageIndex(index)}
-          >
-            <span aria-hidden="true"></span>
-          </button>
-        ))}
-      </div>
+              onClick={() => setImageIndex(index)}
+            >
+              <span aria-hidden="true"></span>
+            </button>
+          ))}
+        </div>
+      )}
     </div>
   );
 };
 
+const ImageCarousel = ({
+  images,
+  auto,
+  interval,
+  hasFullScreen = false,
+  navType = "circles",
+  currentIndex = 0,
+}: ImageCarouselProps) => {
+  const [imageIndex, setImageIndex] = useState(currentIndex);
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const handleImageClick = () => {
+    setIsModalOpen(true);
+  };
+
+  const closeModal = () => {
+    setIsModalOpen(false);
+  };
+
+  useEffect(() => {
+    isModalOpen
+      ? (document.body.style.overflow = "hidden")
+      : (document.body.style.overflow = "scroll");
+  }, [isModalOpen]);
+
+  return (
+    <>
+      <Carousel
+        images={images}
+        auto={auto}
+        interval={interval}
+        navType={navType}
+        imageIndex={imageIndex}
+        imageOnClick={handleImageClick}
+        setImageIndex={setImageIndex}
+      />
+
+      {hasFullScreen &&
+        isModalOpen &&
+        createPortal(
+          <section
+            onClick={closeModal}
+            className="fixed left-0 top-0 z-50 flex h-screen w-screen cursor-auto items-center justify-center overflow-hidden bg-black bg-opacity-75"
+          >
+            <div
+              onClick={(e) => {
+                e.stopPropagation();
+              }}
+              className="relative h-full w-full md:h-[90%] md:w-[60%]"
+            >
+              <button
+                onClick={closeModal}
+                className="absolute bottom-2 left-[45%] z-[55] text-4xl"
```

Comment:
use "left-1/2 transform -translate-1/2" instead of left-[45] to get the button centered exactly

Reply:
button isn't centered exactly with this property. Which is why I went for 45% - however even with this value it isn't haha. 48% seems to do the trick but I'll use the styling you've suggested anyway, as we may go back and change the icon used, depending on Holly's feedback.

Reply:
Oh that's so weird, but yeah we can make note of it for later